From 98e5950c689bf47ed88b12367185ac79b8be6ce7 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Thu, 14 Mar 2024 18:34:05 +0800
Subject: [PATCH 1/4] add flush_queue accel callback

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 etnadrm/etnaviv.c  | 8 ++++++++
 src/armada_accel.h | 1 +
 2 files changed, 9 insertions(+)

diff --git a/etnadrm/etnaviv.c b/etnadrm/etnaviv.c
index 30feba6..262dbd7 100644
--- a/etnadrm/etnaviv.c
+++ b/etnadrm/etnaviv.c
@@ -1046,6 +1046,13 @@ static int etnaviv_export_name(ScreenPtr pScreen, uint32_t name)
 	return fd;
 }
 
+static void etnaviv_flush_queue(ScreenPtr pScreen)
+{
+	struct etnaviv *etnaviv = etnaviv_get_screen_priv(pScreen);
+	if (etnaviv_fence_batch_pending(&etnaviv->fence_head))
+		etnaviv_commit(etnaviv, FALSE);
+}
+
 const struct armada_accel_ops etnaviv_ops = {
 	.pre_init	= etnaviv_pre_init,
 	.screen_init	= etnaviv_ScreenInit,
@@ -1054,5 +1061,6 @@ const struct armada_accel_ops etnaviv_ops = {
 	.attach_name	= etnaviv_attach_name,
 	.free_pixmap	= etnaviv_free_pixmap,
 	.xv_init	= etnaviv_xv_init,
+	.flush_queue	= etnaviv_flush_queue,
 	.export_name	= etnaviv_export_name,
 };
diff --git a/src/armada_accel.h b/src/armada_accel.h
index 073f0d7..73bcb10 100644
--- a/src/armada_accel.h
+++ b/src/armada_accel.h
@@ -33,6 +33,7 @@ struct armada_accel_ops {
 			       unsigned int tv_sec, unsigned int tv_usec,
 			       void *user_data);
 	XF86VideoAdaptorPtr (*xv_init)(ScreenPtr, unsigned int *);
+	void (*flush_queue)(ScreenPtr);
 	int (*export_name)(ScreenPtr, uint32_t);
 };
 
-- 
2.44.0

