From 65efbc42c7ccd5a27ee0e5d9af1303eaa9b155ed Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Sat, 16 Mar 2024 16:27:44 +0800
Subject: [PATCH 2/4] implement present flush

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/common_present.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/common_present.c b/src/common_present.c
index 44b0427..a84b5e7 100644
--- a/src/common_present.c
+++ b/src/common_present.c
@@ -6,6 +6,8 @@
 
 #include "common_drm.h"
 #include "common_drm_helper.h"
+#include "armada_drm.h"
+#include "armada_accel.h"
 
 #include "present.h"
 
@@ -96,7 +98,13 @@ static void common_present_abort_vblank(RRCrtcPtr rr_crtc, uint64_t event_id,
 
 static void common_present_flush(WindowPtr window)
 {
-	/* Flush queued rendering - and wait for it? */
+	ScreenPtr pScreen = window->drawable.pScreen;
+	ScrnInfoPtr pScrn = xf86ScreenToScrn(pScreen);
+	struct armada_drm_info *arm = GET_ARMADA_DRM_INFO(pScrn);
+	const struct armada_accel_ops *ops = arm->accel_ops;
+
+	if (ops)
+		ops->flush_queue(pScreen);
 }
 
 //static Bool common_present_check_flip(RRCrtcPtr crtc, WindowPtr window,
-- 
2.44.0

