From 3e23f839fb11e93ead93784afb792073ee17a1d7 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Fri, 4 Apr 2025 11:26:40 +0800
Subject: [PATCH 6/6] hardcode supported display drivers

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/pvr/wsi/pvr_wsi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pvr/wsi/pvr_wsi.c b/src/pvr/wsi/pvr_wsi.c
index 688059c0337..a88af7e50ed 100644
--- a/src/pvr/wsi/pvr_wsi.c
+++ b/src/pvr/wsi/pvr_wsi.c
@@ -55,6 +55,10 @@ pvr_mesa_wsi_can_present_on_device(VkPhysicalDevice pdevice, int fd)
 #if defined(PVR_DISPLAY_DRIVER_STRING)
       ret = !strcmp(version->name, PVR_DISPLAY_DRIVER_STRING);
 #endif
+      if (!ret)
+         ret = !strcmp(version->name, "es_drm");
+      if (!ret)
+         ret = !strcmp(version->name, "spacemit");
       if (!ret)
          ret = !strcmp(version->name, "pvr");
    }
-- 
2.49.0

