From 9fce9e376b6127e75c6a380ebaf333d9fd7d6ca6 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Mon, 28 Apr 2025 22:27:13 +0800
Subject: [PATCH] HACK: glxext: default to Zink

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/glx/glxext.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/glx/glxext.c b/src/glx/glxext.c
index 72d19a14212..000a719838b 100644
--- a/src/glx/glxext.c
+++ b/src/glx/glxext.c
@@ -885,7 +885,7 @@ __glXInitialize(Display * dpy)
    Bool glx_accel = !debug_get_bool_option("LIBGL_ALWAYS_SOFTWARE", false);
    const char *env = getenv("MESA_LOADER_DRIVER_OVERRIDE");
 
-   zink = env && !strcmp(env, "zink");
+   zink = !env || !strcmp(env, "zink");
    try_zink = False;
 
    dpyPriv->drawHash = __glxHashCreate();
-- 
2.49.0

